addons/stock/models/stock_picking.py

```diff
@@ -566,6 +569,18 @@ def _compute_show_validate(self):
             else:
                 picking.show_validate = True
 
+    @api.depends('state', 'immediate_transfer', 'move_lines', 'move_lines.product_id.tracking', 'move_lines.quantity_done')
+    def _compute_show_set_all_done(self):
+        precision_digits = self.env['decimal.precision'].precision_get('Product Unit of Measure')
+        for picking in self:
+            if (picking.state == 'draft' and picking.immediate_transfer) or picking.state in ('waiting', 'confirmed', 'assigned'):
+                filtered_move_lines = picking.move_lines.filtered(lambda move: move.product_id.tracking == 'none')
+                untracked_products_count = len(filtered_move_lines)
+                done_quantities = sum(filtered_move_lines.mapped('quantity_done'))
```

Comment:
I would use an all(float_is_zero(m.quantity_done,  precision_digits=precision_digits) for m in move_lines_without_tracking)
It avoid browsing the list twice and still easy to read